yarl/_url.py

```diff
@@ -416,9 +418,14 @@ def build(
             )
             fragment = cls._FRAGMENT_QUOTER(fragment) if fragment else fragment
 
-        url = cls._from_val(SplitResult(scheme, netloc, path, query_string, fragment))
         if query:
-            return url.with_query(query)
+            query_string = cls._get_str_query(query) or ""
+
+        url = object.__new__(cls)
+        url._val = tuple.__new__(
+            SplitResult, (scheme, netloc, path, query_string, fragment)
+        )
```

Comment:
this is non-obvious why its much faster.

Could use a comment to point to https://github.com/python/cpython/blob/d83fcf8371f2f33c7797bc8f5423a8bca8c46e5c/Lib/collections/__init__.py#L441